spell/spell-reviewer-mainnet-checklist.md

```diff
@@ -1,369 +1,372 @@
-    * [ ] Exec Doc URL in comment refers to the ['Community' GitHub repo](https://github.com/makerdao/community/tree/master/governance/votes)
-* [ ] Ensure the comments inside the spell action are correct:
-  * [ ] Every _Section text_ from the Exec Sheet should be copied as comment to the spell code (above the code segment that implements the action)
-    * [ ] is surrounded by the set of dashes (E.g. `// ----- Section text -----`)
-  * [ ] Every _Instruction text_ Exec Sheet should be:
-    * [ ] copied to the spell code as `// Instruction text`
-    * [ ] have newline above it
-  * [ ] Every `Reasoning URL` and `Authority URL` from the Exec Sheet should be present under relevant section or instruction in the spell code (depending on which row the link is present)
-    * [ ] Every `Reasoning URL` and `Authority URL` should have prefix derived from the url itself
-      * [ ] `// Executive Vote:` if URL starts with `https://vote.makerdao.com/executive/`
-      * [ ] `// Poll:` if URL starts with `https://vote.makerdao.com/polling/`
-      * [ ] `// Forum:` if URL starts with `https://forum.makerdao.com/t/`
-      * [ ] `// MIP:` if URL starts with `https://mips.makerdao.com/mips/details/`
-  * [ ] If action in the spell doesn't have relevant instruction (e.g.: `chainlog` version bump), it should have explanation prefixed with `// Note:`
-  * [ ] If an instruction can not be taken, it should have a comment under the instruction prefixed with `// Note:` (e.g.: `// Note: Payments are skipped on goerli`)
-* [ ] Local Environment Actions
+  * [ ] Using Exec Doc URL from the above and the target date, generate Exec Doc Hash via `make exec-hash date=$date $URL`
+    _Insert your Exec Doc Hash here_
+  * [ ] Using Exec Doc URL from the above, generate Exec Doc Hash via `cast keccak -- "$(curl '$URL' -o - 2>/dev/null)"`
+    _Insert your Exec Doc Hash here_
+  * [ ] Using Exec Doc URL from the above, read spell instructions from the Exec Doc and list them below
+    _List all instructions announced in the Exec Doc_
+  * [ ] Ensure that instructions announced in the Exec Doc match instructions in the Exec Sheet
+* Base checks
+  * [ ] Current solc version `0.8.16`
+  * [ ] Office hours is `true` IF spell introduces a major change that can affect external parties (e.g.: keepers are affected in case of collateral offboarding) OTHERWISE explicitly set to `false`
+  * [ ] Office hours value matches the Exec Sheet
+  * [ ] Office hours value matches the Exec Doc
+  * [ ] 30 days spell expiry set in the constructor (`block.timestamp + 30 days`)
+* Spell description
+  * [ ] Description follows the format `TARGET_DATE MakerDAO Executive Spell | Hash: EXEC_DOC_HASH`
+  * [ ] Target date in the description matches the Exec Doc target date
+  * [ ] Accompanying comment above spell description follows the format `// Hash: cast keccak -- "$(wget 'EXEC_DOC_URL' -q -O - 2>/dev/null)"`
+  * [ ] Exec Doc URL in comment matches your Raw Exec Doc URL above
```

Comment:
Does the URL needs to be the permanent url? (eg. from specific commit) or the file url in general (eg. https://github.com/makerdao/community/blob/master/governance/votes/Executive%20vote%20-%20January%2024%2C%202024.md) ?

(Because if it is permanent url, I think comparing content hash is enough as some might have different way of getting the url. )

Reply:
Permalink is required for consistency.

Reply:
The underlying problem is that this md document is later edited to include spell address. So if we would use non-permanent url, the hash would change and it will be hard for the external reviewer if some instructions has changed or just the address was added. Therefore, it's required to have permanent url for the hash to make practical sense.

Reply:
Here is the same check in the current version of the checklist:
https://github.com/makerdao/pe-checklists/blob/8e075342d60fac1621673768a76ad797107fbee7/spell/spell-reviewer-mainnet-checklist.md?plain=1#L31

But you're right about different ways of extracting permanent url and about potential mismatches. We probably should formalize one way to avoid potential warning messages. But I think the check is not only about commit hash, but about everything else besides it (eg a malicious actor can share a link to a `makerda0` org which they fully control with a slightly modified copy of the doc which might be easier than getting rights to modify original document).

Anyway, I would put this improvement out of scope as its main purpose is to improve the structure 🙂